ui/app/utils/field-to-attrs.js

```diff
@@ -35,44 +34,45 @@ import { expandProperties } from '@ember/object/computed';
  *
  */
 
-export const expandAttributeMeta = function (modelClass, attributeNames, namePrefix, map) {
-  let fields = [];
+export const expandAttributeMeta = function (modelClass, attributeNames) {
+  const fields = [];
   // expand all attributes
-  attributeNames.forEach((field) => expandProperties(field, (x) => fields.push(x)));
-  let attributeMap = map || new Map();
-  modelClass.eachAttribute((name, meta) => {
-    let fieldName = namePrefix ? namePrefix + name : name;
-    let maybeFragment = get(modelClass, fieldName);
-    if (meta.isFragment && maybeFragment) {
-      // pass the fragment and all fields that start with
-      // the fragment name down to get extracted from the Fragment
-      expandAttributeMeta(
-        maybeFragment,
-        fields.filter((f) => f.startsWith(fieldName)),
-        fieldName + '.',
-        attributeMap
-      );
-      return;
+  // see docs for examples of brace-expansion - https://api.emberjs.com/ember/4.4/functions/@ember%2Fobject%2Fcomputed/expandProperties
+  attributeNames.map((field) => expandProperties(field, (prop) => fields.push(prop)));
+  // cache results of eachAttribute so we don't call it on each iteration of fields loop
```

Comment:
does this mean moving forward we might be able to remove the `_attributeMeta` property we added to some models (like `oidc/scope.js`) to cache `expandAttributeMeta` ?

Reply:
Interesting question :thinking:. I think the issue there is that we only want to expand the attribute meta once, whereas if we don't cache the initial result the getter will call this method every time. I think if we want to continue to use a getter for the form fields we will have to stick with that pattern.